packages/@aws-cdk/aws-rds/test/instance.test.ts

```diff
@@ -1583,6 +1583,24 @@ describe('instance', () => {
     });
   });
 
+  test('throws with backupRetention on a read replica if engine does not support it', () => {
+    // GIVEN
+    const instanceType = ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.SMALL);
+    const backupRetention = cdk.Duration.days(5);
+    const source = new rds.DatabaseInstance(stack, 'Source', {
+      engine: rds.DatabaseInstanceEngine.postgres({ version: rds.PostgresEngineVersion.VER_13 }),
+      backupRetention,
+      instanceType,
+      vpc,
+    });
+
+    expect(() => new rds.DatabaseInstanceReadReplica(stack, 'Replica', {
+      sourceDatabaseInstance: source,
+      backupRetention,
+      instanceType,
+      vpc,
+    })).toThrow(/Cannot set `backupRetention` for a postgres read replica/);
```

Comment:
```suggestion
    expect(() =>{
      new rds.DatabaseInstanceReadReplica(stack, 'Replica', {
        sourceDatabaseInstance: source,
        backupRetention,
        instanceType,
        vpc,
      });
    }).toThrow(/Cannot set `backupRetention` for a postgres read replica/);
```

(Sorry, GutHub's UI seems screwed up today, I can't select lines correctly)

Reply:
Done. A note on this `{ }` style for `toThrow` expectations: it allows to "easily" add statements inside the `{ }` which is not a good practice when testing that a single statement should throw.

```ts
expect(() => myFun()).toThrow(); // cannot easily add statements inside the throw expectation
```

```ts
expect(() => {
  myFun(); // can easily add statements inside (above or below)
}).toThrow();
```

😄

Reply:
You are correct, but for me, it's more important that I see what's being tested o(`myFun()` in this case) on a separate line 🙂.

Reply:
OK. The change has been made.